### PR TITLE
enhancement(vrl): allow statements in `if` predicates over multiple lines.

### PR DESCRIPTION
**Pull request by StephenWakely**
This PR makes it possible to specify the statements in `if` predicates to work over multiple lines, so the following becomes possible:

```
if (
  x = thing
  y= thung
  x == y 
) {
...
}
```

I've noticed some users have fairly lengthy predicates and this change makes things more ergonomic for them.

### CODE REVIEW

File: lib/vrl/parser/src/parser.lalrpop [hunk 1]
```diff
@@ -445,9 +445,13 @@ ElseIf: IfStatement =
...
 };
 
+StatementSeparator: () = {
+    NonterminalNewline, ";"
+};
+
 Predicate: Predicate = {
     Box<ArithmeticExpr> => Predicate::One(<>),
-    "(" <v:(<AssignmentExpr> ";")+> <e:(<AssignmentExpr>)?> ")" => {
+    "(" StatementSeparator? <v:(<AssignmentExpr> StatementSeparator)+> <e:(<AssignmentExpr>)?> ")" => {
```
Comment (JeanMertz) [inline lib/vrl/parser/src/parser.lalrpop:454]: The first `StatementSeparator?` should be `NonterminalNewline*`

Also, I don't believe this makes it possible to have multiple newlines between expressions, which would be required if someone wanted to write:

```
if (
	# here's my foo check
	foo

	# and here's the bar check
	bar
) { ... }
```

**Review by JeanMertz — CHANGES_REQUESTED** [review]

**Comment by binarylogic** [comment]
@StephenWakely just a reminder we need to update the docs for these types of changes:

https://github.com/vectordotdev/vector/blob/master/website/cue/reference/remap/expressions/if.cue

File: lib/vrl/tests/tests/expressions/if_statement/multiline_predicates.vrl [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  true
+}
+
+v4 = if (
+  # Lets test spaces
+  x = x + 1
+
+  # Does it work?
+
+  x = x + 1; x == 7;
```
Comment (JeanMertz) [inline lib/vrl/tests/tests/expressions/if_statement/multiline_predicates.vrl:27]: This final `;` shouldn't be needed.

**Review by JeanMertz — CHANGES_REQUESTED** [review]
One more issue, but otherwise this looks good.

File: lib/vrl/tests/tests/expressions/if_statement/multiline_predicates.vrl (hunk 2 above)
Comment (JeanMertz) [reply lib/vrl/tests/tests/expressions/if_statement/multiline_predicates.vrl:27]: In fact, we should disallow it, as it means the `x == 7` expression shouldn't be used as the resulting expression for the predicate. Instead, the predicate should resolve to `null`, given that there isn't any expression after `;`.

Comment (StephenWakely) [reply lib/vrl/tests/tests/expressions/if_statement/multiline_predicates.vrl:27]: As discussed, I am finding this to be quite a tricky thing to do, it brings up all sorts of ambiguity issues in lalrpop. I have tested on master and this is an issue there as well, so this is not a regression. I would like to defer this and have raised #9255 to track this.

**Review by JeanMertz — APPROVED** [review]

**Pull request merged**